src/Workspaces/Core/Portable/Options/GlobalOptionService.cs

```diff
@@ -58,20 +112,103 @@ private object LoadOptionFromSerializerOrGetDefault(OptionKey optionKey)
 
         public IEnumerable<IOption> GetRegisteredOptions()
         {
-            return _options.Value;
+            return _lazyAllOptions.Value;
         }
 
+        public ImmutableHashSet<IOption> GetRegisteredSerializableOptions(ImmutableHashSet<string> languages)
+        {
+            if (languages.IsEmpty)
+            {
+                return ImmutableHashSet<IOption>.Empty;
+            }
+
+            var builder = ImmutableHashSet.CreateBuilder<IOption>();
+
+            // "string.Empty" for options from language agnostic option providers.
+            builder.AddRange(GetOrComputeSerializableOptionsForLanguage(string.Empty));
+
+            foreach (var language in languages)
+            {
+                builder.AddRange(GetOrComputeSerializableOptionsForLanguage(language));
+            }
+
+            return builder.ToImmutable();
+        }
+
+        /// <summary>
+        /// Gets force computed serializable options with prefetched values for all the registered options applicable to the given <paramref name="languages"/> by quering the option persisters.
+        /// </summary>
+        public SerializableOptionSet GetForceComputedOptions(ImmutableHashSet<string> languages, IOptionService optionService)
```

Comment:
>GetForceComputedOptions [](start = 37, length = 23)

Not clear what "force" means here. Would `GetSerializableOptionsSnapshot` name make sense for this method?